gcloud-java-bigquery/src/main/java/com/google/gcloud/bigquery/Dataset.java

```diff
@@ -0,0 +1,285 @@
+
+  private final BigQuery bigquery;
+  private final DatasetInfo info;
+
+  private static class TablePageFetcher implements PageImpl.NextPageFetcher<Table> {
+
+    private static final long serialVersionUID = 6906197848579250598L;
+
+    private final BigQueryOptions options;
+    private final Page<BaseTableInfo> infoPage;
+
+    TablePageFetcher(BigQueryOptions options, Page<BaseTableInfo> infoPage) {
+      this.options = options;
+      this.infoPage = infoPage;
+    }
+
+    @Override
+    public Page<Table> nextPage() {
+      Page<BaseTableInfo> nextInfoPage = infoPage.nextPage();
+      return new PageImpl<>(new TablePageFetcher(options, nextInfoPage),
+          nextInfoPage.nextPageCursor(), new LazyTableIterable(options, nextInfoPage.values()));
+    }
+  }
+
+  private static class LazyTableIterable implements Iterable<Table>, Serializable {
+
+    private static final long serialVersionUID = 3312744215731674032L;
+
+    private final BigQueryOptions options;
+    private Iterable<BaseTableInfo> infoIterable;
+    private transient BigQuery bigquery;
+
+    public LazyTableIterable(BigQueryOptions options, Iterable<BaseTableInfo> infoIterable) {
+      this.options = options;
+      this.infoIterable = infoIterable;
+      this.bigquery = options.service();
+    }
+
+    private void readObject(ObjectInputStream in) throws IOException, ClassNotFoundException {
+      in.defaultReadObject();
+      this.bigquery = options.service();
+    }
+
+    @Override
+    public Iterator<Table> iterator() {
+      return Iterators.transform(infoIterable.iterator(), new Function<BaseTableInfo, Table>() {
+        @Override
+        public Table apply(BaseTableInfo tableInfo) {
+          return new Table(bigquery, tableInfo);
+        }
+      });
+    }
+
+    @Override
+    public int hashCode() {
+      return Objects.hash(options, infoIterable);
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+      if (!(obj instanceof LazyTableIterable)) {
+        return false;
+      }
+      LazyTableIterable other = (LazyTableIterable) obj;
+      return Objects.equals(options, other.options)
+          && Objects.equals(infoIterable, other.infoIterable);
+    }
+  }
+
+  /**
+   * Constructs a {@code Dataset} object for the provided {@code DatasetInfo}. The BigQuery service
+   * is used to issue requests.
+   *
+   * @param bigquery the BigQuery service used for issuing requests
+   * @param info dataset's info
+   */
+  public Dataset(BigQuery bigquery, DatasetInfo info) {
+    this.bigquery = checkNotNull(bigquery);
+    this.info = checkNotNull(info);
+  }
+
+  /**
+   * Creates a {@code Dataset} object for the provided dataset's user-defined id. Performs an RPC
+   * call to get the latest dataset information.
+   *
+   * @param bigquery the BigQuery service used for issuing requests
+   * @param dataset dataset's user-defined id
+   * @return the {@code Dataset} object or {@code null} if not found.
```

Comment:
Nit, though consistent I wonder where it comes from (the '.' for `@return`).
The [Javastyle](http://www.oracle.com/technetwork/java/javase/documentation/index-137868.html#@return) guide suggests it should be treated like the `@param` tag.